### PR TITLE
feat: implement querying for commit hash and proofs

### PR DESCRIPTION
**Pull request by p0mvn**
**Context**
https://github.com/osmosis-labs/osmosis/issues/1137

Sample Query:
```
}root@roman-relayer:~# curl -X GET "localhost:26657/abci_query?path=\"%2Fstore%2Fproofs\"&height=3738243" -H "accept: application/json"
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "response": {
      "code": 0,
      "log": "",
      "info": "",
      "index": "0",
      "key": "cHJvb2Zz",
      "value": "ehl2s43Sj+klwt5pKFjf4/pYvbNKejpT+z3LHILeGMM=",
      "proofOps": {
        "ops": [
          {
            "type": "ics23:simple",
            "key": "ZGlzdHJpYnV0aW9u",
            "data": "CoQCCgxkaXN0cmlidXRpb24SIKx7Dz+kx8ZLy/B4eZJUbtfqEK6p6wmUM8T5VcSTNmshGgkIARgBIAEqAQAiJwgBEgEBGiCgO5TgM6JBoSDQOEkW5flnF/dyAMpRSaut6pVEVTJvqSIlCAESIQGsAfi6pJMSzJaHc+Wjlf20mpYh8OIkbNMoKX88Q0pmACIlCAESIQG1/9gZ/96CAVXCYq6NXTp7lJ913MFPYoGGacnykPepoSInCAESAQEaIEW5aviA028cgbdfSQCmrIWKBOSnbpD4A/4CJKD5R/fuIicIARIBARogNkzkfbC0gqwiknjCkIHw9qQ0QlYi9CW85pmvi69Pdwg="
          },
           ...
        ]
      },
      "height": "3738243",
      "codespace": ""
    }
  }
```

### CODE REVIEW

File: store/rootmulti/store.go [hunk 1]
```diff
@@ -567,19 +570,37 @@ func (rs *Store) getStoreByName(name string) types.Store {
 // TODO: add proof for `multistore -> substore`.
```
Comment (ValarDragon) [inline store/rootmulti/store.go]: Lets update the godoc for the `proofs` special case?

File: store/rootmulti/store.go [hunk 2]
```diff
@@ -567,19 +570,37 @@ func (rs *Store) getStoreByName(name string) types.Store {
...
+			Height: req.Height,
+			Key: []byte(proofsPath),
+			Value: commitInfo.CommitID().Hash,
+			ProofOps: &crypto.ProofOps{Ops: make([]crypto.ProofOp, 0, len(commitInfo.StoreInfos))},
+		}
+
+		for _, storeInfo := range commitInfo.StoreInfos {
+			res.ProofOps.Ops = append(res.ProofOps.Ops, commitInfo.ProofOp(storeInfo.Name))	
+		}
+		return res
```
Comment (ValarDragon) [inline store/rootmulti/store.go]: Can we move this all to a separate function?

**Review by ValarDragon — APPROVED** [review]
LGTM, two minor comments!

File: store/rootmulti/store.go (hunk 1 above)
Comment (p0mvn) [reply store/rootmulti/store.go]: Wdym by that exactly? Updating the method comment or anything else?

`godoc` seems to be deprecated: https://pkg.go.dev/golang.org/x/tools/cmd/godoc

Comment (p0mvn) [reply store/rootmulti/store.go]: Going to merge, will update this in another PR if I didn't get it right

Comment (ValarDragon) [reply store/rootmulti/store.go]: Oh, it just means adding a comment about the function for any exported token.

Theres sites that auto-compile this into a docs site. https://godocs.io/github.com/osmosis-labs/osmosis

**Pull request merged**